containers/src/Kokkos_OffsetView.hpp

```diff
@@ -1883,160 +1883,145 @@ struct MirrorOffsetType {
 
 }  // namespace Impl
 
-template <class T, class... P>
+namespace Impl {
+template <class T, class... P, class... I>
 inline typename Kokkos::Experimental::OffsetView<T, P...>::HostMirror
-create_mirror(
-    const Kokkos::Experimental::OffsetView<T, P...>& src,
-    typename std::enable_if<
-        !std::is_same<typename Kokkos::ViewTraits<T, P...>::array_layout,
-                      Kokkos::LayoutStride>::value>::type* = nullptr) {
-  using src_type = Experimental::OffsetView<T, P...>;
-  using dst_type = typename src_type::HostMirror;
-
-  return dst_type(
-      Kokkos::Impl::ViewCtorProp<std::string>(
-          std::string(src.label()).append("_mirror")),
-      typename Kokkos::ViewTraits<T, P...>::array_layout(
-          src.extent(0), src.extent(1), src.extent(2), src.extent(3),
-          src.extent(4), src.extent(5), src.extent(6), src.extent(7)),
+create_mirror(const Kokkos::Experimental::OffsetView<T, P...>& src,
+              const I&... arg_prop) {
+  return typename Kokkos::Experimental::OffsetView<T, P...>::HostMirror(
+      Kokkos::create_mirror(arg_prop..., src.view()), src.begins());
+}
+
+template <class Space, class T, class... P, class... I>
+inline typename Kokkos::Impl::MirrorOffsetType<Space, T, P...>::view_type
+create_mirror(const Space&,
+              const Kokkos::Experimental::OffsetView<T, P...>& src,
+              const I&... arg_prop) {
+  return typename Kokkos::Impl::MirrorOffsetType<Space, T, P...>::view_type(
+      Kokkos::view_alloc(src.label(), arg_prop...), src.layout(),
```

Comment:
Is switching from constructing `array_layout` to passing `src.layout()` also expected to handle non-unit strides, where it didn't previously?

Reply:
I would need to evaluate that.

Reply:
The constructor called has the same implementation as the one for `Kokkos::View` so the answer should be"Yes, it allows non-unit strides".